pyk/src/pyk/kcfg/kcfg.py

```diff
@@ -684,6 +732,90 @@ def remove_node(self, node_id: NodeIdLike) -> None:
         for alias in [alias for alias, id in self._aliases.items() if id == node_id]:
             self.remove_alias(alias)
 
+    def remove_node_safely(
+            self,
+            node_id: NodeIdLike,
+            predecessors: dict[str, tuple[NodeIdLike, ...]] = None,
+            successors: dict[str, tuple[NodeIdLike, ...]] = None,
+    ) -> None:
```

Comment:
Although the current remove node operation does not cause dangling issues, it directly removes edges related to the node, leading to some unexpected situations. For example:
```mermaid
graph LR
A -->|Split| B
A -->|Split| C
```
Removing node B results in:
```mermaid
graph LR
A 
C
```
However, what I expect is:
```mermaid
graph LR
A -->|Split| C
```

Reply:
This is actually expected behavior, because `Split` nodes must maintain that they are _total_ and _mutually exclusive_. Mutually exclusive means that nodes `B` and `C` must not have any intersection of their states. Total means that node `A` must represent the same set of states as the union of nodes `B` and `C`. So putting a split like `A -> C` violates the totalness requirement, so the split must be removed.